README.md
<!-- 
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages). 

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages). 
-->

Envision Fency container library used for different type of widgets like, Gradiant, Imagebackground or Profile detail containers. This make easy to your coding life.

## Features

This package inclue the 
Container with gradiant background.
Container with image background.
Container with profile detail.

## Getting started

Conatiner with Gradiant
en_fency_container(
                  width: 300,
                  title: "Title", 
                  subTitle: "Some detail info here...", 
                  height: 100,
                  color1: Colors.yellow, 
                  color2: Colors.red, 
                  textColor: Colors.white, 
                  type: containerType.gradient,),

Container with image backgourd
en_fency_container(
                  imageUrl: 'https://images.unsplash.com/photo-1610307766999-01f45d259959?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
                  width: 300,
                  type: containerType.imageBackground, 
                  title: "Title 1", 
                  subTitle: "Some detail info here...", 
                  textColor: Colors.white, 
                  height: 250,),

Profile setting with detail
en_fency_container(
                  profileImgUrl: 'https://dl.memuplay.com/new_market/img/com.vicman.newprofilepic.icon.2022-06-07-21-33-07.png',
                  userName: "MR ANDY",
                  userEmail: 'andy@gmail.com',
                  userPhoneNo: '+123456789876',
                  height: 100, 
                  width: 300,
                  type: containerType.profile,
                  borderColor: Colors.blueGrey,)


## Example

https://tinyurl.com/55tpmh6n



CHANGELOG.md
## 1.0.6

* This is an library for use fency containers with text, image, backgourd, profile. 


test/en_fency_container_test.dart
// ignore_for_file: unused_import

import 'package:flutter_test/flutter_test.dart';

import 'package:en_fency_container/en_fency_container.dart';

void main() {
  test('adds one to input values', () {
    
  });
}


lib/src/en_fency_container.dart
// ignore_for_file: prefer_const_constructors, camel_case_types, unused_element, dead_code

import 'package:en_fency_container/src/imageBackgroud_container_widget.dart';
import 'package:en_fency_container/src/profile_container_widget.dart';
import 'package:flutter/material.dart';
import 'package:en_fency_container/src/gradiant_container_widget.dart';

enum containerType {
    gradient,
    imageBackground,
    profile,
}

class en_fency_container extends StatefulWidget {
  const en_fency_container({Key? key, this.width, this.height, this.title, this.subTitle, this.color1, this.color2, this.textColor, this.type, this.imageUrl, this.borderColor, this.profileImgUrl, this.userEmail, this.userName, this.userPhoneNo}) : super(key: key);
  final double? width;
  final double? height;
  final String? title;
  final String? subTitle;
  final Color? color1;
  final Color? color2;
  final Color? textColor;
  final containerType? type;
  final String? imageUrl;
  final Color? borderColor;
  final String? profileImgUrl;
  final String? userName;
  final String? userEmail;
  final String? userPhoneNo;

  @override
  State<en_fency_container> createState() => _en_fency_containerState();
}

class _en_fency_containerState extends State<en_fency_container> {
  @override
  Widget build(BuildContext context) {
    return container();
  }

  Widget container(){
        if (widget.type == containerType.gradient){
            return GradiantContainerWidget(widget: widget);
        }else if (widget.type == containerType.imageBackground){
            return imageBackgroudContainerWidget(widget: widget);
        }else if (widget.type == containerType.profile){
            return profileContainerWidget(widget: widget);
        }

        return container();

    }
}



lib/src/profile_container_widget.dart
import 'package:flutter/material.dart';

import '../en_fency_container.dart';

class profileContainerWidget extends StatelessWidget {
  const profileContainerWidget({
    Key? key,
    required this.widget,
    
  }) : super(key: key);

  final en_fency_container widget;

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: EdgeInsets.all(10),
      width: widget.width ?? MediaQuery.of(context).size.width * 0.90,
      height: widget.height ?? 120,
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(10),
        border: Border.all(color: widget.borderColor ?? Colors.black)
      ),
      child: Row(
        children: [
          Image.network(widget.profileImgUrl ?? 'https://as1.ftcdn.net/v2/jpg/03/46/83/96/1000_F_346839683_6nAPzbhpSkIpb8pmAwufkC7c5eD7wYws.jpg', width: 100, height: 100,),
          Column(
            children: [
              Text("${widget.userName}", 
                style: TextStyle(
                    color: widget.textColor, 
                    fontSize: 20, 
                    fontWeight: FontWeight.bold,),
                maxLines: 2,
                textAlign: TextAlign.center,),
              Text("${widget.userEmail}", 
                style: TextStyle(
                    color: widget.textColor, 
                    fontSize: 14, 
                    fontWeight: FontWeight.normal),
                    textAlign: TextAlign.center,),
              Text("${widget.userPhoneNo}", 
                style: TextStyle(
                    color: widget.textColor, 
                    fontSize: 14, 
                    fontWeight: FontWeight.normal),
                    textAlign: TextAlign.center,),
            ],
          ),
        ],
      ),
    );
  }
}


lib/src/gradiant_container_widget.dart
import 'package:flutter/material.dart';
import 'en_fency_container.dart';

class GradiantContainerWidget extends StatelessWidget {
  const GradiantContainerWidget({
    Key? key,
    required this.widget,
    
  }) : super(key: key);

  final en_fency_container widget;

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: EdgeInsets.all(10),
      width: widget.width ?? MediaQuery.of(context).size.width * 0.90,
      height: widget.height ?? 120,
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(10),
        gradient: LinearGradient(
          colors: [
              widget.color2??Colors.blue, 
              widget.color2??Colors.red
            ],
          begin: Alignment.topLeft,
          end: Alignment.bottomRight,
          stops: const [0.1, 0.5],
        ),
      ),
      child: Column(
        children: [
          Text("${widget.title}", 
            style: TextStyle(
                color: widget.textColor, 
                fontSize: 20, 
                fontWeight: FontWeight.bold,),
            maxLines: 2,
            textAlign: TextAlign.center,),
          Text("${widget.subTitle}", 
            style: TextStyle(
                color: widget.textColor, 
                fontSize: 14, 
                fontWeight: FontWeight.normal),
                textAlign: TextAlign.center,),
        ],
      ),
    );
  }
}

lib/src/imageBackgroud_container_widget.dart
// ignore_for_file: prefer_const_constructors

import 'package:flutter/material.dart';

import '../en_fency_container.dart';

class imageBackgroudContainerWidget extends StatelessWidget {
  const imageBackgroudContainerWidget({
    Key? key,
    required this.widget,
    
  }) : super(key: key);

  final en_fency_container widget;

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: EdgeInsets.all(10),
      width: widget.width ?? MediaQuery.of(context).size.width * 0.90,
      height: widget.height ?? 120,
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(10),
      ),
      child: Stack(
        children: [
            Image.network(widget.imageUrl ?? "https://picsum.photos/250?image=9"),
            Column(
          children: [
            Text("${widget.title}", 
              style: TextStyle(
                  color: widget.textColor, 
                  fontSize: 20, 
                  fontWeight: FontWeight.bold,),
              maxLines: 2,
              textAlign: TextAlign.center,),
            Text("${widget.subTitle}", 
              style: TextStyle(
                  color: widget.textColor, 
                  fontSize: 14, 
                  fontWeight: FontWeight.normal),
                  textAlign: TextAlign.center,),
          ],
        )],
      ),
    );
  }
}

example/lib/main.dart
// ignore_for_file: prefer_const_constructors, prefer_const_literals_to_create_immutables

import 'dart:ffi';

import 'package:flutter/material.dart';
import 'package:en_fency_container/en_fency_container.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Padding(
          padding: const EdgeInsets.all(8.0),
          child: ListView(
            children: [
              en_fency_container(
                  width: 300,
                  title: "Title", 
                  subTitle: "Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.",
                  color1: Colors.yellow, 
                  color2: Colors.red, 
                  textColor: Colors.white, 
                  type: containerType.gradient,),
              SizedBox(height: 20,),
              en_fency_container(
                  imageUrl: 'https://images.unsplash.com/photo-1610307766999-01f45d259959?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
                //   width: 300,
                  type: containerType.imageBackground, 
                  title: "Title 1", 
                  subTitle: "Some detail info here...", 
                  textColor: Colors.white, 
                  height: 250,),
              SizedBox(height:20),
              en_fency_container(
                  profileImgUrl: 'https://dl.memuplay.com/new_market/img/com.vicman.newprofilepic.icon.2022-06-07-21-33-07.png',
                  userName: "MR ANDY",
                  userEmail: 'andy@gmail.com',
                  userPhoneNo: '+123456789876',
                  height: 100, 
                  width: 300,
                  type: containerType.profile,
                  borderColor: Colors.blueGrey,)
            ],
          ),
        )
      ),
      // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
